### ISSUE TITLE
Expand Releases Alert Options (Session Alerts)

### ISSUE THREAD

**Issue by kpujjigit** [getsentry/sentry#76113]
### Problem Statement

Currently it's possible to build dashboards around count_healthy(), count_abnormal(), and count_crashed() metrics that can pass in users or sessions. Allowing users to construct alerts based on healthy or abnormal session count can allow for end users more granularity when choosing to deprecate or modify old releases. 

### Solution Brainstorm

Enable the aforementioned counts in the Alert wizard to construct session alerts.

![Image](https://github.com/user-attachments/assets/0bee2fca-18d6-45ec-acde-029a59fed346)

### Product Area

Alerts

**Comment by gggritso** [comment]
@rachrwang @kpujjigit sorry I'm a little confused, is this about Alerts, Dashboards, or both? If I'm reading this right, it looks like Dashboards already support this, but Alerts do not, is that right? If yes, I think this is for the Alerts team rather than Dashboards.

**Comment by kpujjigit** [comment]
@gggritso You are right, this is intended for Alerting team. Seems Santry incorrectly routed this.

**Comment by nhsiehgit** [comment]
Makes sense
there's a WIP big overhaul of Alerts currently under way that will be changing a lot of how the system works.
In the future world, one goal will be to enable any/every query from the discover/insights tools (or whatever tool you can create a query with) to then be translated into an identical alert

**Comment by rene-leanix** [comment]
@nhsiehgit, is there any update on the overhaul of alerts you mentioned and the possibility of creating session-based alerts for other than crash-free sessions?

The reason this would be very important for us too is the following: we have a browser single-page-application consisting of over 40 micro-front-ends, which are continuously released, leading to 200 and more releases per day. Each of these releases is only gradually adopted as users refresh their browsers or reach a page that requires the load of a new micro-front-end. The ability to monitor the health of these releases and to be alerted on high error rates is crucial.

We do have alerts on new errors for each of those micro-front-ends, which already helps a lot, especially during working hours, as long as teams are quick enough to check out the error & session rates for those errors manually. But what we really need is an alert that notifies us if the rate of _error_-free (not _crash_-free) sessions of a new release is dropping below a certain threshold. We cannot achieve the same with alerts that are based on absolute error counts, as those may trigger too late, in case a release is slowly adopted.

Our workaround at the moment is to mark most errors we report to Sentry as `fatal`, which causes the corresponding sessions to be marked as `crashed`, so we can use the built-in crash-free session rate alert. But this isn't ideal, as those errors were actually handled by our code and could be recovered from.

If the generic solution you mentioned is a long way off, would it be possible to add an "error-free" session rate alert in addition to the "crash-free" session rate alert or to allow adding an additional level-filter option for the latter?

**Comment by ameliahsu** [comment]
Thanks for following up on this feature request! This work is in our team's backlog to be scheduled. In the meantime, we are currently working on improvements to our alerting that will enable adding an additional level-filter option, as you have suggested. Feel free to ping on this ticket again for status updates.
